src/repr/src/row.proto

```diff
@@ -79,7 +79,6 @@ enum ProtoDatumOther {
     FALSE = 2;
     TRUE = 3;
     JSON_NULL = 4;
-    DUMMY = 5;
```

Comment:
The best practice for removing a proto enum variant or message field is to move the id part to be "reserved" to prevent it from being accidentally being reused: https://protobuf.dev/programming-guides/proto3/#reserved-values. We control all prod instances, so it's not strictly necessary if we're sure it's not written down anywhere, but the false negative case is so confusing to debug I think we should do it anyway (we get 127 of these variant ids before they affect the encoded size)